tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc

```diff
@@ -4489,16 +4489,21 @@ class ConvertTensorScatterOp : public OpRewritePattern<OpTy> {
         op.updates().getType().template dyn_cast<RankedTensorType>();
 
     if (!tensor_ty || !indices_ty || !updates_ty) return failure();
-    // Last dimension of the indices needs to known at compile time for
-    // computation of the 'update_window_dims' attribute in the dimensions
-    // struct.
-    int64_t num_index_dims = indices_ty.getShape().back();
-    if (ShapedType::isDynamic(num_index_dims)) return failure();
 
     int64_t tensor_rank = tensor_ty.getRank();
     int64_t indices_rank = indices_ty.getRank();
     int64_t updates_rank = updates_ty.getRank();
 
+    // Handling special case with rank0 and rank1 indices, making them rank2
+    std::vector<int64_t> indices_shape = indices_ty.getShape();
+    if(indices_rank == 0) indices_shape = {1,1};
```

Comment:
The op requires indices of rank one at least.
https://github.com/tensorflow/tensorflow/blob/6c34306c0f5fcb56b1f02c74381a338429b853da/tensorflow/core/ops/array_ops.cc#L3084

Reply:
okok, so does this mean I can remove the check for rank == 0?

Reply:
done